client/src/components/Auth/Login/__tests__/Login.test.jsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import { makeMockStore } from "../../../../test/testUtils";
+import { render, cleanup, fireEvent } from "@testing-library/react";
+import { renderWithRouter } from "../../../../test/routerTestUtils";
+import { Redirect as MockRedirect } from "react-router-dom";
+import API from "../../../../lib/api";
+import Login from "../Login";
+
+const defaultProps = {
+  auth: {
+    isAuthenticated: false,
+    user: {},
+    loading: false
+  }
+};
+
+jest.mock("react-router-dom", () => {
+  const original = jest.requireActual("react-router-dom");
+  return {
+    ...original,
+    Redirect: jest.fn().mockImplementation(() => null)
+  };
+});
```

Comment:
Would it make sense to create a `__mocks__` folder at the top-level directory so this is always mocked? Or is that going too far?

Reply:
probably too far. this mock is only used for this file so far.

Reply:
actually, this might even get refactored anyway.